src/dstore/pmix_esh.c

```diff
@@ -144,23 +175,251 @@ static inline const char *_unique_id(void)
     return str;
 }
 
+static inline int _esh_nspace_tbl_init(void)
+{
+    _esh_nspace_cnt = ESH_INIT_NSPACE_CNT;
+    _esh_nspace_tbl_size = ESH_INIT_NSPACE_TBL_SIZE;
+
+    _nspace_tbl = (nspace_tbl_entry_t*)malloc(sizeof(nspace_tbl_entry_t) * _esh_nspace_tbl_size);
+    if (NULL == _nspace_tbl) {
+        PMIX_ERROR_LOG(PMIX_ERROR);
+        return PMIX_ERROR;
+    }
+    memset(_nspace_tbl, 0x00, sizeof(nspace_tbl_entry_t) * _esh_nspace_tbl_size);
+
+    _nspace_map = (nspace_entry_t*)malloc(sizeof(nspace_entry_t) * _esh_nspace_cnt);
+    if (NULL == _nspace_map) {
+        PMIX_ERROR_LOG(PMIX_ERROR);
```

Comment:
Use more detaile `PMIX_ERR_OUT_OF_RESOURCE` errcode instead of universal `PMIX_ERROR`